docs/resource/k8s.md

```diff
@@ -354,3 +354,70 @@ A CronJob creates Jobs on a repeating schedule.
 <!-- endsemconv -->
 
 [DocumentStatus]: https://opentelemetry.io/docs/specs/otel/document-status
+
+## Specify resource attributes using Kubernetes annotations
+
+All annotations with the `resource.opentelemetry.io/` prefix should be translated into the corresponding resource
+attributes.
+
+For example, the annotation `resource.opentelemetry.io/service.name` should be translated into the `service.name`
+attribute.
+
+## Service
+
+The following [service resource attributes](../attributes-registry/service.md) are recommended for Kubernetes services.
+
+There are different ways to calculate the service attributes.
+
+1. [Well-Known Labels](https://kubernetes.io/docs/reference/labels-annotations-taints/)
+2. Annotations on the pod template that have the `resource.opentelemetry.io/` prefix as described above
+3. A function of the Kubernetes resource attributes defined above
+
+This translation can typically be done by an OpenTelemetry Collector component like the
+[Kubernetes Attribute Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor).
+
+Tools offering this functionality should provide an opt-in flag for the use of well-known labels,
+since users may not be aware that their labels are being used for this purpose.
+
+Each attribute has a priority order for how it should be calculated as described below.
+
+### How `service.namespace` is calculated
+
+Choose the first value found:
+
+1. `pod.annotation[resource.opentelemetry.io/service.namespace]`
+2. `pod.label[app.kubernetes.io/part-of]` (well-known label
+   [app.kubernetes.io/part-of](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-part-of))
+3. `k8s.namespace.name`
+
+### How `service.name` is calculated
+
+Choose the first value found:
+
+- `pod.annotation[resource.opentelemetry.io/service.name]`
+- `pod.label[app.kubernetes.io/name]` (well-known label
+  [app.kubernetes.io/name](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-name))
+- `k8s.deployment.name`
+- `k8s.replicaset.name`
+- `k8s.statefulset.name`
+- `k8s.daemonset.name`
+- `k8s.cronjob.name`
+- `k8s.job.name`
+- `k8s.pod.name`
+- `k8s.container.name`
+
+### How `service.version` is calculated
+
+Choose the first value found:
+
+- `pod.annotation[resource.opentelemetry.io/service.version]`
+- `pod.label[app.kubernetes.io/version]` (well-known label
+  [app.kubernetes.io/version](https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-version))
+- `if (contains(container docker image tag, '/') == false) container docker image tag`
+
+### How `service.instance.id` is calculated
+
+Choose the first value found:
+
+- `pod.annotation[resource.opentelemetry.io/service.instance.id]`
+- `concat([k8s.namespace.name, k8s.pod.name, k8s.container.name], '.')`
```

Comment:
We discussed `service.instance.id` in the semconv meeting and had some doubts about the scenario where a container is restarted without a pod recreate (e.g. kubelet restarting a container for OOM). The confusion was whether this id should change to reflect that a new instance of the container was started which is what would happen i think if we left this to an sdk.

Reply:
The convention for `service.instance.id` states that it `MUST be unique for each instance of the same service.namespace,service.name pair (in other words service.namespace,service.name,service.instance.id triplet MUST be globally unique).` For a pod which is owned by a statefulset this is not the case. I guess the question is whether this id should be used to differentiate between different instances of a service at a given time or across all instances ever run. In case of a statefulset, the id is not unique enough to differentiate between instances of a pod with same sts ordinal but k8s will only run one instance of a pod with a given ordinal (the id will remain same after an sts pod is recreated).

Reply:
> IMO this is somewhat moot since SDKs are going to start generating unique IDs for service.instance.id anyways.
Are sdks going to ignore any `service.instance.id` values passed in with the env var `OTEL_RESOURCE_ATTRIBUTES`? My understanding is that the documented values are currently being calculated and set in this var by operator when auto-instrumenting.

Reply:
> > IMO this is somewhat moot since SDKs are going to start generating unique IDs for service.instance.id anyways.
> 
> Are sdks going to ignore any `service.instance.id` values passed in with the env var `OTEL_RESOURCE_ATTRIBUTES`? My understanding is that the documented values are currently being calculated and set in this var by operator when auto-instrumenting.

no, SDKs must honor whatever attributes are set using `OTEL_RESOURCE_ATTRIBUTES`

Reply:
> I've generally interpreted "globally unique" here to mean unique at a given time, rather than unique across all time.

that is also my understanding

Reply:
thanks @jsuereth for the feedback on this question, which I have now included in the PR itself:

> Note that the container restart count is not included in the `service.instance.id` calculation, because it makes
troubleshooting harder when the ID changes on every restart, e.g. in a crash loop.